config/formats.js

```diff
@@ -271,6 +271,93 @@ let Formats = [
 		ruleset: ['[Gen 7] Doubles OU'],
 		banlist: ['DOU', 'DBL'],
 	},
+	{
+		name: "[Gen 7] VGC 2019 (Sun)",
+
+		mod: 'gen7',
+		gameType: 'doubles',
+		forcedLevel: 50,
+		teamLength: {
+			validate: [4, 6],
+			battle: 4,
+		},
+		timer: {starting: 6 * 60 + 30 - 10, perTurn: 10, maxPerTurn: 55, maxFirstTurn: 90, timeoutAutoChoose: true, dcTimerBank: false},
+		ruleset: ['Pokemon', 'Minimal GBU'],
+		banlist: ['Unown', 'Rayquaza + Dragon Ascent'],
+		requirePlus: true,
+		onValidateTeam: function (team) {
+			const legends = ['Mewtwo', 'Lugia', 'Ho-Oh', 'Kyogre', 'Groudon', 'Rayquaza', 'Dialga', 'Palkia', 'Giratina', 'Reshiram', 'Zekrom', 'Kyurem', 'Xerneas', 'Yveltal', 'Zygarde', 'Cosmog', 'Cosmoem', 'Solgaleo', 'Lunala', 'Necrozma'];
+			let n = 0;
+			let problems = [];
+			for (const set of team) {
+				const baseSpecies = this.getTemplate(set.species).baseSpecies;
+				if (legends.includes(baseSpecies)) n++;
+				if (n > 2) {
+					problems.push(`You can only use up to two legendary Pok\u00E9mon.`);
+					n = -6; // Prevent the message from being listed more than once.
```

Comment:
Perhaps put the legends count check at the end? (You would also have the option of suffixing `(You have ${n} legendary Pok\u00E9mon.)`.)